report/renderer/report-ui-features.js

```diff
@@ -339,13 +339,15 @@ export class ReportUIFeatures {
   }
 
   /**
-   * DevTools uses its own file manager to download files, so it redefines this function.
-   * Wrapper is necessary so DevTools can still override this function.
-   *
    * @param {Blob|File} blob
    */
   _saveFile(blob) {
-    const filename = getLhrFilenamePrefix(this.json);
-    this._dom.saveFile(blob, filename);
+    const ext = blob.type.match('json') ? '.json' : '.html';
```

Comment:
The flow report won't include a file extension automatically now, can you update:

https://github.com/GoogleChrome/lighthouse/blob/ad0824dea1dc796da6fde188609ee6dc0129cdc6/flow-report/src/topbar.tsx#L17-L21